starlite/openapi/controller.py

```diff
@@ -95,12 +127,25 @@ def retrieve_schema_yaml(self, request: Request) -> "OpenAPI":
                 A [Request][starlite.connection.Request] instance.
 
         Returns:
-            A rendered YAML object..
+            resoponse: With a rendered YAML object
         """
-        return self.get_schema_from_request(request)
+        if not request.app.openapi_config:  # pragma: no cover
+            raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
+
+        if self.should_serve_endpoint(request):
+            return Response(
+                content=self.get_schema_from_request(request),
+                status_code=HTTP_200_OK,
+                media_type=OpenAPIMediaType.OPENAPI_YAML,
+            )
+        return Response(
```

Comment:
ok, here the response should not be of type HTML. It should be of type `OpenAPIMediaType.OPENAPI_YAML` with content being an empty dictionary.